backends/kafka/lag.go

```diff
@@ -0,0 +1,160 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/batchcorp/plumber/cli"
+	"github.com/batchcorp/plumber/printer"
+	"github.com/pkg/errors"
+	skafka "github.com/segmentio/kafka-go"
+)
+
+// validate cli options and init connection
+func Lag(opts *cli.Options) error {
+	if err := validateLagOptions(opts); err != nil {
+		return errors.Wrap(err, "unable to validate read options")
+	}
+
+	kafkaLag, err := NewKafkaLagConnection(opts)
+
+	for _, v := range kafkaLag.partitionDiscoverConn {
+		defer v.Close()
+	}
+
+	if err != nil {

```

Comment:
This error check should go directly after the `NewKafkaLagConnection()` call, otherwise the for loop will panic if the connection failed and `kafkaLag` is nil

Reply:
resolving in the next commit